### PR TITLE
[Feature]: Make generating TestReports files optional 

### PR DESCRIPTION
**Pull request by Akash-Singh04**
## Related Issue
  - Info about Issue or bug

Closes: #1393 

#### Describe the changes you've made
Added a Flag ```--generateTestReports``` to ```keploy test```  ```keploy serve``` and config file that will allow the user to choose if they want to generate the Test Reports or not, which is set to true by default. If the flag is set to false via ```generateTestReport=false``` in CLI or config, the folder will be deleted at the end of the CLI execution.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

NIL

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

image

### CODE REVIEW

File: cmd/test.go [hunk 1]
```diff
@@ -107,6 +107,8 @@ type Test struct {
 	logger *zap.Logger
 }
 
+var deleteTestReport bool
```
Comment (gouravkrosx) [inline cmd/test.go:110]: Take flag name as `generateTestReport` which should be true by default.

File: pkg/service/generateConfig/generateConfig.go [hunk 2]
```diff
@@ -70,6 +70,7 @@ test:
         ports: 0
   withCoverage: false
   coverageReportPath: ""
+  generateTestReport: false
```
Comment (gouravkrosx) [inline pkg/service/generateConfig/generateConfig.go:73]: By default, it should be true.

Comment (gouravkrosx) [reply pkg/service/generateConfig/generateConfig.go:73]: And where are you reading this `generateTestReport` flag from the config file?

**Comment by gouravkrosx** [comment]
@Akash-Singh04 Please address the comments.

File: cmd/test.go (hunk 1 above)
Comment (Akash-Singh04) [reply cmd/test.go]: Changed flag name and set default value to true

The implementation is now as follows:

**Comment by Akash-Singh04** [comment]
@gouravkrosx Comments have been addressed, Kindly review and revert

**Pull request closed without merging**